layers/ntp.go

```diff
@@ -0,0 +1,450 @@
+//     |                                                               |
+//     .                                                               .
+//     .                    Extension Field 2 (variable)               .
+//     .                                                               .
+//     |                                                               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |                          Key Identifier                       |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |                                                               |
+//     |                            dgst (128)                         |
+//     |                                                               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     From http://www.ietf.org/rfc/rfc5905.txt
+//
+// The fields "Extension Field 1 (variable)" and later are optional fields,
+// and so we can set a minimum NTP record size of 48 bytes.
+//
+const ntp_minimum_record_size_in_bytes int = 48
+
+//******************************************************************************
+
+// NTP Type
+// --------
+// Type NTP implements the DecodingLayer interface. Each NTP object
+// represents in a structured form the NTP record present as the UDP
+// payload in an NTP UDP packet.
+//
+type NTP struct {
+	BaseLayer // Stores the packet bytes and payload bytes.
+
+	LeapIndicator uint8 // [0,3]. Indicates whether leap second(s) is to be added.
+	Version       uint8 // [0,7]. Version of the NTP protocol.
+	Mode          uint8 // [0,7]. Mode.
+	Stratum       uint8 // [0,255]. Stratum of time server in the server tree.
+	Poll          int8  // [-128,127]. The maximum interval between
+	//             successive messages, in log2 seconds.
+	Precision int8 // [-128,127]. The precision of the system clock,
+	//             in log2 seconds.
+	RootDelay uint32 // [0,2^32-1]. Total round trip delay to the reference clock
+	//             in seconds times 2^16.
+	RootDispersion uint32 // [0,2^32-1]. Total dispersion to the reference clock,
+	//             in seconds times 2^16.
+	ReferenceID        uint32 // ID code of reference clock [0,2^32-1].
+	ReferenceTimestamp uint64 // Most recent timestamp from the reference clock.
+	OriginTimestamp    uint64 // Local time when request was sent from local host.
+	ReceiveTimestamp   uint64 // Local time (on server) that request arrived at server host.
+	TransmitTimestamp  uint64 // Local time (on server) that request departed server host.
+
+	// FIX: This package should analyse the extension fields and represent the extension fields too.
+	ExtensionBytes []byte // Just put extensions in a byte slice.
+}
+
+//******************************************************************************
+
+// LayerType() returns the layer type of the NTP object, which is LayerTypeNTP.
```

Comment:
Follow https://blog.golang.org/godoc-documenting-go-code for all docs.  IE: here, change to:

```go
// LayerType returns the layer type of the NTP object, which is LayerTypeNTP.
func (d *NTP) LayerType() gopacket.LayerType {
```

IE: remove () after function name, remove empty comment line (line 313)